### PR TITLE
chore(notification): no more typehint errors when sending a notifiction as an ElggGroup

### LINKED ISSUE #7949: incorrect type hint in Elgg_Notifications_Event
"Argument 3 passed to Elgg_Notifications_Event::__construct() must be an instance of ElggUser, instance of ElggGroup given, called in engine/lib/notification.php on line 462 and defined" in file engine/classes/Elgg/Notifications/Event.php (line 34)

if notify_user is called with the from guid of something else than an ElggUser (which is allowed according to the function documentation) this error get triggered as the constructor of Elgg_Notifications_Event type hints to ElggUser.

Maybe this should be changed to ElggEntity?

### PR DESCRIPTION
**Pull request by jeabakker**
When calling notify_user with the sender as a group the
Elgg\Notifications\Event class would trigger a typehint warning that is
expects an ElggUser

fixes #7949

### CODE REVIEW

File: engine/classes/Elgg/Notifications/Event.php [hunk 1]
```diff
@@ -55,7 +55,7 @@ public function __construct(\ElggData $object, $action, \ElggUser $actor = null)
 	/**
 	 * Get the actor of the event
 	 *
-	 * @return \ElggUser
+	 * @return \ElggEntity
```
Comment (hypeJunction) [inline engine/classes/Elgg/Notifications/Event.php]: In theory |false

**Comment by hypeJunction** [comment]
LGTM if tests are passing

**Comment by jdalsem** [comment]
LGTM if you address the comment of @hypeJunction and fix the typo in the commit message "notification"

**Comment by jeabakker** [comment]
updated commit

**Pull request merged**